commit 6d134fb4defecb16591adbf4acb020acd165a75a
Author: Paul Jakma <paul.jakma@sun.com>
Date:   Thu Aug 23 23:22:02 2007 +0000

    [bgpd] Pass NOSUB to regexec
    
    2007-08-23 Paul Jakma <paul.jakma@sun.com>
    
    	* bgp_regex.c: (bgp_regcomp) Pass NOSUB flag to regcomp to
    	  prevent parsing of substitutions, which can have profound
    	  performance effects on bgpd and are of no use to the CLI
    	  anyway. How much it helps depends on the regex
    	  implementation.

diff --git a/bgpd/bgp_regex.c b/bgpd/bgp_regex.c
index be84d40..9b65f7c 100644
--- bgpd/bgp_regex.c
+++ bgpd/bgp_regex.c
@@ -66,7 +66,7 @@ bgp_regcomp (const char *regstr)
 
   regex = XMALLOC (MTYPE_BGP_REGEXP, sizeof (regex_t));
 
-  ret = regcomp (regex, magic_str, REG_EXTENDED);
+  ret = regcomp (regex, magic_str, REG_EXTENDED|REG_NOSUB);
 
   XFREE (MTYPE_TMP, magic_str);
 
